### PR TITLE
[15.0][IMP] stock_picking_auto_create_lot: Compute quantities to reservation for moves which lots will be created in picking validation

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa TT42995

ping @carlosdauden @CarlosRoca13

### CODE REVIEW

**Review by AungKoKoLin1997 — APPROVED** [review]
Functional Review: It works as expected.

File: stock_picking_auto_create_lot/models/stock_move.py [hunk 1]
```diff
@@ -22,3 +22,22 @@ def _compute_display_assign_serial(self):
...
+        """
+        res = super()._set_quantities_to_reservation()
+        moves_to_process = self.filtered(
+            lambda mv: not mv.quantity_done
+            and mv.picking_type_id.use_create_lots
+            and mv.picking_type_id.auto_create_lot
+            and mv.product_id.auto_create_lot
+        )
+        for move in moves_to_process:
+            if move.state not in ("partially_available", "assigned"):
```
Comment (yostashiro) [inline stock_picking_auto_create_lot/models/stock_move.py]: I think this condition can be moved inside `filtered()` or the content of `filtered()` can be moved inside this for loop.

**Review by CarlosRoca13 — APPROVED** [review]
LGTM :+1:

**Review by yostashiro — COMMENTED** [review]
I suppose this PR shares the same background as https://github.com/OCA/stock-logistics-workflow/pull/1283?

**Comment by pedrobaeza** [comment]
@sergio-teruel can you check latest comment?

**Comment by sergio-teruel** [comment]
Changes done!

**Comment by pedrobaeza** [comment]
Merging as no further comments from @yostashiro 

/ocabot merge patch

**Pull request merged**